stat/distmat/permutation.go

```diff
@@ -0,0 +1,53 @@
+// Copyright ©2020 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package distmat
+
+import (
+	"golang.org/x/exp/rand"
+
+	"gonum.org/v1/gonum/mat"
+)
+
+// UniformPermutation is a distribution over permutation matrices
```

Comment:
```
// UniformPermutation is a distribution over uniform permutation matrices.
```

Reply:
I'm being very pedantic here as I needed to google that term.  Seems "uniform permutation matrix" is used, very occasionally in a handful of references, to mean "random permutation matrix uniformly chosen among all the N! choices." (https://arxiv.org/pdf/1111.4662.pdf)

So that change is fine as far as it goes.  But I'm unlikely to be the only person checking to see if there are other special properties (there are not).  A style question I guess on the vocabulary.

Reply:
If there is convention, use that. It may be worth providing a link if there is work on this that supports it.

Reply:
changed to: a uniform distribution over the n! permutation matrices of size n x n for a given n.  i can find no widely-used  naming and expect that statement is clear to the audience using gonum.